src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementService.java

```diff
@@ -132,6 +137,45 @@ private void makePayableResourceIfUnpaid(String createdUri,
         }
     }
 
+    public ValidationStatusResponse areTasksComplete(String submissionId) throws CompanyNotFoundException {
+        Optional<ConfirmationStatementSubmissionJson> submissionJsonOptional = getConfirmationStatement(submissionId);
+        if (submissionJsonOptional.isEmpty()) {
+            throw new CompanyNotFoundException();
```

Comment:
CompanyNotFoundException is probably not the correct one to throw here. Maybe use a ServiceException with a message indicating why it failed. Or if you want a specific one, create a SubmissionNotFound one?